docs/design/environment/enviroment_management.md

```diff
@@ -0,0 +1,153 @@
+# Kusion Environment Management Proposal
+
+## Context
+
+Towards the version 0.10.0, Kusion is about to make a huge upgrade: the unambiguous definition of responsibility and collaboration paradigm between platform engineers and application developers through the application delivery process. The platform engineers firstly define modules and corresponding application-oriented schemas, where the latter constitute the application schema [AppConfiguration](https://www.kusionstack.io/docs/user_docs/config-walkthrough/overview#appconfiguration-model), and secondly initialize environments and corresponding configurations. The application developers select environment and fill in the fields of AppConfiguration. Then, executing the delivery action in a trusted environment, e.g. a CD pipeline, to apply the application intent from both the platform engineers and application developers. 
+
+The upgrade defines the role of platform engineers and the concepts of **Module** and **Environment**. The Module is a building block enabling the delivery of a set of resources from **Schema Configuration** and **Environment Configuration**. While the Environment is a collection of application-independent configurations, and an environment configuration is composed of a set of module inputs, which forms the entire module inputs combined with schema configuration.
+
+<br />![workflow](workflow.png)
+
+## Goals and Non-Goals
+
+### Goals
+
+1. Give the definition of environment; 
+2. Define the writing format of environment configurations;
+3. Provide the commands to manage environments;
+4. Give the integration scheme with AppConfiguration when running `kusion build`.
+
+### Non-Goals
+
+1. Only support local file based environment management, the support of other storage backends is not the goal of version 0.10.0;
+2. Only consider using Kusion in CLI format, the usage of service layer is not considered.
+
+## Proposal
+
+### Definition
+
+Environment is a logical concept representing a target that a stack will be deployed to, which contains a set of configurations, kubeconfig and provider authentication information that can be reused by multiple stacks. We recommend environments grouped by SDLC phases or cloud vendors. For example, it could be named as dev, staging, and prod or cloud vendors such as AWS, Azure, and Aliyun.
```

Comment:
> it could be named as dev, staging, and prod or cloud vendors such as AWS, Azure, and Aliyun

or could be both

Reply:
You're tight, an environment could be like AWS_dev